stdlib/builtin/regexp.rbs

```diff
@@ -109,6 +112,10 @@ class Regexp < Object
 
   def self.quote: (String | Symbol arg0) -> String
 
+  def self.union: () -> self
```

Comment:
This test told me the existence of a bug. 🐛 

Please use `Regexp` instead of `self`. `self` is intended to be the type of the receiver, so in this case it should be resolved to `singleton(Regexp)`. Currently, it is resolved to `Regexp`, and this is a bug.

Reply:
Thank you for your kind review! 😄 
I fixed it in https://github.com/ruby/ruby-signature/pull/75/commits/4ff33bc1906355bda3c53d36a3989c1ef3be62bc!

Since `self` was also used in `::compile`, I fixed both `::compile` and `::union`!